docs/usage.md

```diff
@@ -40,3 +40,37 @@ Note that Ruff's pre-commit hook should run before Black, isort, and other forma
 
 Ruff can also be used as a [VS Code extension](https://github.com/charliermarsh/ruff-vscode) or
 alongside any other editor through the [Ruff LSP](https://github.com/charliermarsh/ruff-lsp).
+
+Ruff can also be used as a [GitHub Action](https://github.com/features/actions) :
+
+You can use Ruff within a GitHub Actions workflow without setting your own Python environment. Great for enforcing your Ruff rules.  Naturally, assumes the repo has an appropriate [config](https://beta.ruff.rs/docs/configuration/) in a `pyproject.toml`, `ruff.toml`, or `.ruff.toml`.
+
+Compatibility
+This action is known to support all GitHub-hosted runner OSes. In addition, only published versions of Ruff are supported (i.e. whatever is available on PyPI).
+
+Usage
+Create a file named .github/workflows/ruff.yml inside your repository with:
+
+```yaml
+name: Ruff
+
+on: [push, pull_request]
+
+jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: charliermarsh/ruff@v0
+```
+
+The version of Ruff the action will use can be configured via version. This can be any valid version specifier or just the version number if you want an exact version. The action defaults to the latest release available on PyPI. Only versions available from PyPI are supported, so no commit SHAs or branch names.
+
+You can also configure the arguments passed to Ruff via options (defaults to 'check) and src (default is '.').
+
+```yaml
+- uses: charliermarsh/ruff@v0
+  with:
+    src: "./src"
+    version: "0.0.259"
+```
```

Comment:
Do you mind adding one example that passes a command-line option to Ruff? (E.g. `--select B`, just as an arbitrary example.)

Reply:
FYI ... many different formats for version work.

`version: v0.0.259`
`version: "v0.0.259"`
`version: 0.0.259`
`version: "0.0.259`

I tend to think to put things in quotes that are to be treated as a string which is why I use it there.

Reply:
added as suggestion

Reply:
and now committed